contracts/vault/src/execute.rs

```diff
@@ -309,66 +304,68 @@ fn execute_collect_yield(
         .add_attribute("markets_processed", markets.len().to_string()))
 }
 
-fn execute_process_withdrawal(deps: DepsMut, env: Env, _info: MessageInfo) -> Result<Response> {
+pub fn execute_process_withdrawal(deps: DepsMut, env: Env, _info: MessageInfo) -> Result<Response> {
     let config = state::CONFIG.load(deps.storage)?;
-    check_not_paused(&config)?;
-    // No authorization check - open to all users to process pending withdrawals
-    // Load the withdrawal queue
-    let mut queue = state::WITHDRAWAL_QUEUE.load(deps.storage)?;
 
-    // Check if there are any pending withdrawals
-    if queue.is_empty() {
-        return Err(anyhow!("No pending withdrawals to process"));
+    if config.paused {
+        return Err(anyhow!("El contrato está pausado"));
     }
 
     let vault_balance = deps
         .querier
         .query_balance(&env.contract.address, &config.usdc_denom)?
         .amount;
+
     let mut messages: Vec<CosmosMsg> = vec![];
     let mut processed_amount = Uint128::zero();
     let limit = 20;
+    let mut processed_ids: Vec<QueueId> = vec![];
+
+    for item in state::WITHDRAWAL_QUEUE
+        .range(deps.storage, None, None, Order::Ascending)
+        .take(limit)
+    {
+        let (id, request) = item?;
+
+        match processed_amount.checked_add(request.amount) {
```

Comment:
This is fine to leave as it is but realistically we will never overflow here and even if we did, the `Add` impl for `Uint128` will actually panic instead of silently wrapping around. So can simply use `processed_amount + request.amount`.